src/ServiceInsight/MessageFlow/MessageNode.cs

```diff
@@ -39,20 +48,27 @@ public MessageNode(MessageFlowViewModel owner, StoredMessage message)
             });
         }
 
-        string ProcessSagaType(StoredMessage message)
+        IEnumerable<SagaInfo> ProcessSagaType(StoredMessage message)
         {
             if (message.Sagas == null)
             {
-                return string.Empty;
+                yield break;
             }
 
-            var originatingSaga = message.Sagas.FirstOrDefault();
-            if (originatingSaga == null)
+            foreach (var saga in message.Sagas)
             {
-                return string.Empty;
+                var originatingSaga = saga;
```

Comment:
Is this still the right name for this variable?

Reply:
Probably not. I'll rename.